tutorials/micro/micro_onnx.py

```diff
@@ -0,0 +1,209 @@
+relay_mod = relay.transform.DynamicToStatic()(relay_mod)
+
+######################################################################
+# Next, we lower the Relay model to the specific device we are
+# targeting. In this case, we are using the Nordic Semiconductor
+# `nRF5340DK development board <https://www.nordicsemi.com/Software-and-tools/Development-Kits/nRF5340-DK>`.
+
+# This is the device target name used by microTVM. It is used to select default
+# options for the device when we use ``tvm.target.target.micro()`` below.
+UTVM_TARGET = "nrf5340dk"
+
+# This is the board designation used by Zephyr, and required for the compilation process.
+UTVM_ZEPHYR_BOARD = "nrf5340dk_nrf5340_cpuapp"
+
+######################################################################
+# If you wish to run against an emulated Zephyr device using QEMU,
+# you can uncomment these lines instead:
+# UTVM_TARGET = "host"
+# UTVM_ZEPHYR_BOARD = "qemu_x86"
+
+
+######################################################################
+# We define the TVM target here.
+# We add -link-params=1 option here, so that the model parameters are
+# included in the resulting binary image.
+target = tvm.target.target.micro(UTVM_TARGET, options=["-link-params=1"])
+
+######################################################################
+# Now, we do the Relay build.
+TVM_OPT_LEVEL = 3
+with tvm.transform.PassContext(opt_level=TVM_OPT_LEVEL, config={"tir.disable_vectorize": True}):
+    lowered = relay.build(relay_mod, target, params=params)
+    graph_json_str = lowered.get_json()
+
+######################################################################
+# Next, create a uTVM Workspace. This is a location where the
+# generated Zephyr project will be compiled.
+workspace_root = os.path.abspath(
+    f'workspace/{datetime.datetime.now().strftime("%Y-%m-%dT%H-%M-%S")}'
+)
+workspace_parent = os.path.dirname(workspace_root)
+if not os.path.exists(workspace_parent):
+    os.makedirs(workspace_parent)
+workspace = tvm.micro.Workspace(debug=True, root=workspace_root)
+print(f"Using workspace: {workspace_root}")
+
+######################################################################
+# Now we create the ``ZephyrCompiler`` object, which generates the
+# Zephyr project for hosting the device runtime code, as well as
+# generating the device-specific binary from our model.
+
+# You are welcome to implement your own Zephyr-based runtime environment
+# for your project. This runtime is a demo which provides basic capabilities
+# for interfacing to the microTVM device code via the device's serial
+# port.
+UTVM_ZEPHYR_RUNTIME_DIR = "../../apps/microtvm/zephyr/demo_runtime/"
+
+# The ``west`` command is used by Zephyr for compilation and
+# flashing devices.
+UTVM_WEST_CMD = "west"
+
+compiler = zephyr.ZephyrCompiler(
+    project_dir=UTVM_ZEPHYR_RUNTIME_DIR,
+    board=UTVM_ZEPHYR_BOARD,
+    zephyr_toolchain_variant="zephyr",
```

Comment:
Zephyr SDK is not available on Mac OS. Maybe we can use arm toolchain <gnuarmemb> for now?

Reply:
I'm following this: https://docs.zephyrproject.org/latest/guides/beyond-GSG.html#gs-toolchain
Still trying to figure it out.

Reply:
For the nRF5340, I used this:

https://infocenter.nordicsemi.com/topic/ug_gsg_ncs/UG/gsg/install_toolchain.html
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/gs_installing.html

- Launch Toolchain Manager from nRF Connect
- Once installed, select "Launch IDE"
- Open a project, e.g., the Zephyr samples "Blinky"
- Make sure that "Project 'zephyr/zephyr.elf'" is highlighted in the projects panel
- Build > Build Solution
- Target > Connect J-Link
- Target > Download zephyr/zephyr.elf

Reply:
@mdw-octoml thanks for the information. I just checked this tutorial with "zephyr" sdk on VM and it works fine.